adaptive/learner/new_learnerND.py

```diff
@@ -758,27 +166,30 @@ def _finalize_initialization(self):
         for subdomain in self.domain.subdomains():
             self.queue.insert(subdomain, priority=self.priority(subdomain))
 
-        self._initialized = True
-
     def priority(self, subdomain):
         if self._initialized:
-            L_0 = self.loss_function(self.domain, subdomain, self.codomain_bounds, self.data)
+            if subdomain in self.losses:
+                L_0 = self.losses[subdomain]
+            else:
+                L_0 = self.loss_function(
+                    self.domain, subdomain, self.codomain_bounds, self.data
+                )
+                self.losses[subdomain] = L_0
         else:
-            # Before we're initialized we don't have enough data to calculate losses,
-            # so we just assign the same loss to each subdomain
-            L_0 = 1
+            # Before we have all the boundary points we can't calculate losses because we
+            # do not have enough data. We just assign the subdomain volume as the loss.
+            L_0 = self.domain.volume(subdomain)
+
         subvolumes = self.domain.subvolumes(subdomain)
-        # We use this tuple as the priority, and we also store the loss directly so that
-        # we can easily look it up later.
-        return ((max(subvolumes) / sum(subvolumes)) * L_0, L_0)
+        return (max(subvolumes) / sum(subvolumes)) * L_0
 
     def ask(self, n, tell_pending=True):
         if self.n_asked >= len(self.boundary_points):
             points, losses = self._ask(n, tell_pending)
         else:
-            points = self.boundary_points[self.n_asked:self.n_asked + n]
+            points = self.boundary_points[self.n_asked : self.n_asked + n]
             # The boundary points should always be evaluated with the highest priority
-            losses = [float('inf')] * len(points)
+            losses = [float("inf")] * len(points)
```

Comment:
The [canonical place](https://docs.python.org/3/library/math.html#math.inf) for infinity is `math.inf`.

Reply:
fixed